mlflow/utils/search_utils.py

```diff
@@ -9,12 +9,16 @@
 class SearchFilter(object):
     VALID_METRIC_COMPARATORS = set(['>', '>=', '!=', '=', '<', '<='])
     VALID_PARAM_COMPARATORS = set(['!=', '='])
+    VALID_TAG_COMPARATORS = set(['!=', '='])
```

Comment:
Have been thinking some about this. One of the things we should look at search is treat the following as the same:

```
params.model = "LR"
PARAMS.model = "LR"
Params.model = "LR"
```

This is outside the scope of this PR, but if we are looking at this as a SQL search query: It makes sense to have values and column names are case-sensitive but can we relax this constraint for table names?